pyrsa/data/dataset.py

```diff
@@ -0,0 +1,165 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Definition of RSA Dataset class and subclasses
+@author: baihan, jdiedrichsen
+"""
+
+import numpy as np
+import pyrsa as rsa
+
+
+class DatasetBase:
+    """
+    Abstract dataset class.
+    Defines members that every class needs to have, but does not
+    implement any interesting behavior. Inherit from this class
+    to define specific dataset types
+
+        Args:
+            measurements (numpy.ndarray): n_obs x n_channel 2d-array,
+                                          or n_set x n_obs x n_channel
+                                          3d-array
+            descriptors (dict):           descriptors with 1 value per
+                                          Dataset object
+            obs_descriptors (dict):       observation descriptors (all
+                                          are array-like with shape =
+                                          (n_obs,...))
+            channel_descriptors (dict):   channel descriptors (all are
+                                          array-like with shape =
+                                          (n_channel,...))
+        Returns:
+            dataset object
+    """
```

Comment:
Maybe the init for the dataset objects should do some checks that the descriptors are all compatible with the size of the measurement matrix?

Reply:
Will include this in the next few pushes. Thanks!

Reply:
> For all functions of the dataset (split, subset etc.) you currently only return the measurements. Shouldn't this be new dataset objects with descriptors attached?

fixed and pushed.

Reply:
fixed and pushed.